### DISCUSSION TITLE
Stats tab ideas

### DISCUSSION THREAD

**Discussion by matthew73210** [tphakala/birdnet-go#157] [General]
Hey! What are your plans regarding the stats tab?
Are you gunning for a juypterlab esq interactive env [Gophernotes](https://github.com/gopherdata/gophernotes)?
The full monty with [go-charts](https://github.com/go-echarts/go-echarts)?

I like the idea of the interactive side, not sure how it fits with your ethos of this app though.

Cheers

**Comment by tphakala** [comment]
I would like to keep things rather simple to use for now, once there is basic charting built maybe we could think of more advanced options. 

For charting I have been eyeing [go-charts](https://github.com/vicanso/go-charts) and [go-echarts](https://github.com/go-echarts/go-echarts), go-echarts seems to have more active development but it does not have suitable dark theme out of box, but maybe it could be added easily.

For simplicity I think [go-charts](https://github.com/vicanso/go-charts) would be a good to start with.

Few type of statistics could include
- Daily active hours per species and as total
- Species occurrence over time on weekly and monthly scale
- Species occurrence with weather correlation (this requires weather data from OpenWeatherMap)

Anything else?

**Comment by matthew73210** [comment]
I kinda like the idea of a 3 graph plotting birds, time, days or something of the sort.

Regarding themes, i think you can set what ever backgroup colour you want. Might even be able to have it dynamic ?!
Saying that echarts seems the way to go.

Had a dig in the files, (still not to familiar with go, kinda reminds me og C++ back in my uni days) and in datastore you seem to have a couple functions that can serve up data. 
"GetAllNotes" seems like the easy way to go for the gophernotes (might have a go)
"SpeciesDetections" seems like the best bet.

Next weekend i'll see if i can spin something simple up based on [echarts](https://go-echarts.github.io/go-echarts/#/)

**Comment by matthew73210** [comment]
Some reading : https://zetcode.com/golang/echarts/
